numpy/core/setup.py

```diff
@@ -605,6 +606,12 @@ def generate_api(ext, build_dir):
     config.add_include_dirs(join('src', 'umath'))
     config.add_include_dirs(join('src', 'npysort'))
 
+    # Need to add location of npy_math_internal.h, generated from src/npymath/npy_math_internal.h.src
+    npymath_build_dir = join('build', 'src.%s-%s.%s' % (sysconfig.get_platform(), sys.version_info[0], sys.version_info[1]),
```

Comment:
interesting these functions seem to be cruft from a removed compilation mode.

But still explicitly setting the build directory should not be necessary, npysort which is a static utility library simlar to npymath has a generated headers, `npy_partition.h.src`. Have you tried just adding the .src file to the sources list and nothing else?